backend/de.metas.async/src/main/java/de/metas/async/processor/impl/planner/AsyncProcessorPlanner.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * #%L
+ * de.metas.async
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.async.processor.impl.planner;
+
+import com.google.common.collect.ImmutableSet;
+import de.metas.async.model.I_C_Queue_WorkPackage;
+import de.metas.async.processor.IQueueProcessor;
+import de.metas.async.processor.impl.AbstractQueueProcessor;
+import de.metas.async.processor.impl.ThreadPoolQueueProcessor;
+import de.metas.logging.LogManager;
+import lombok.NonNull;
+import org.adempiere.util.concurrent.CustomizableThreadFactory;
+import org.slf4j.Logger;
+
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class AsyncProcessorPlanner extends QueueProcessorPlanner
+{
+	private static final Logger logger = LogManager.getLogger(QueueProcessorPlanner.class);
+
+	private ExecutorService plannerExecutorService;
+	private ExecutorService queueProcessorExecutorService;
+
+	public AsyncProcessorPlanner()
+	{
+		super();
+
+		this.plannerExecutorService = null;
+		this.queueProcessorExecutorService = null;
+	}
+
+	protected void startPlanner()
+	{
+		restartExecutorsIfTerminated();
+
+		isRunning.set(true);
```

Comment:
why is this `isRunning.set(true)` in between the other 2 methods?

same question for `shutdownPlanner` method

Reply:
it doesn't have any real impact, just a matter of how I thought about things